test/utils/typed_value.hpp

```diff
@@ -0,0 +1,29 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+#pragma once
+
+#include "types.hpp"
+#include "value.hpp"
+
+namespace fizzy::test
+{
+struct TypedValue
+{
+    Value value;
+    ValType type;
+
+    /// Assign type to the typeless Value.
+    /// No validation is possible, so the correctness is on the user only.
+    constexpr TypedValue(ValType ty, Value v) noexcept : value{v}, type{ty} {}
+
+    constexpr TypedValue(int32_t v) noexcept : TypedValue{ValType::i32, v} {}
+    constexpr TypedValue(uint32_t v) noexcept : TypedValue{ValType::i32, v} {}
+    constexpr TypedValue(int64_t v) noexcept : TypedValue{ValType::i64, v} {}
+    constexpr TypedValue(uint64_t v) noexcept : TypedValue{ValType::i64, v} {}
+    constexpr TypedValue(float v) noexcept : TypedValue{ValType::f32, v} {}
+    constexpr TypedValue(double v) noexcept : TypedValue{ValType::f64, v} {}
+
+    constexpr operator Value() const noexcept { return value; }
```

Comment:
Where is this used?

Reply:
I don't know any more. Some future PRs propose to drop it.

Reply:
It was used ones, I dropped it. It is always easy to drop the type information by `.value`.

Reply:
Yeah, I was concerned that with it can be misleading e.g.
```
TypedValue v{1_u64};
if (v == Value{1_i32}) // true
```